blog/client.ts

```diff
@@ -0,0 +1,69 @@
+import type {
+  PostFrontMatterWithSlug,
+  PostFrontMatterCollection,
+} from "./models";
+import { FRONT_MATTER_CACHE_FILENAME, POSTS_BUILD_DIR } from "./paths";
+import { extractImports } from "./extractImports";
+import { type Mdx, parseMdx } from "./parser";
+
+export async function getPosts(origin: string) {
+  let url: URL;
+  if (process.env.NODE_ENV === "development") {
+    url = new URL(`http://localhost:8080/${FRONT_MATTER_CACHE_FILENAME}`);
+  } else {
+    url = new URL(FRONT_MATTER_CACHE_FILENAME, origin);
+  }
+  return fetch(url).then((response) => {
+    if (!response.ok) {
+      throw new Error(`Could not retrieve posts. ${response.statusText}`);
+    }
+    return response.json() as Promise<PostFrontMatterCollection>;
+  });
+}
+
+export async function getPost(origin: string, slug: string): Promise<Mdx> {
+  let url!: URL;
+  if (process.env.NODE_ENV === "development") {
+    url = new URL(`http://localhost:8080/posts/${slug}.mdx`);
+    return fetch(url).then((response) => {
+      if (!response.ok) {
+        throw new Error(`Could not retrieve post. ${response.statusText}`);
+      }
+      return response.json();
+    });
+  } else {
```

Comment:
I'll explore using a resource route this week. I noticed some weird behavior when trying to read the file system in dev mode from the Remix server. The `__dirname` of the root is actually the `build` folder and not the folder where the source files are actually found. I'm guessing because these files are compiled to `build` before executing. If I can find a way to cleanly read from the file system in both scenarios, then a resource route is definitely the end of the separate dev server.